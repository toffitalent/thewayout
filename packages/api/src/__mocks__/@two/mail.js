const send = jest.fn().mockResolvedValue(undefined);

module.exports = {
  send,
};
