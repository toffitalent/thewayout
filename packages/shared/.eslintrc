{
  "settings": {
    "import/resolver": {
      "typescript": {
        "project": ["tsconfig.json", "packages/shared/tsconfig.json"]
      }
    }
  },
  "rules": {
    "no-param-reassign": "off",
    "react/react-in-jsx-scope": "off"
  }
}
