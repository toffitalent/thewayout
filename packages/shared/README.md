# @two/shared

Copyright (c) 2023-Present The Way Out, Inc. All Rights Reserved.

## Development

This package contains code shared between the [api](../api/README.md), [app](../app/README.md), and [web](../web/README.md) packages and is not intended to run in a standalone manner. See the READMEs of those packages for information on starting a development environment.

## Testing

To run all component and unit tests and get coverage reports, simply run `yarn workspace @dlatwobs/shared test`.
