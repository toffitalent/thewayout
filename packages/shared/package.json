{
  "name": "@two/shared",
  "version": "1.0.0",
  "author": "Disruptive Labs",
  "license": "UNLICENSED",
  "private": true,
  "main": "build/index.js",
  "types": "build/index.d.ts",
  "scripts": {
    "build": "tsc -b tsconfig.build.json",
    "clean": "rimraf build coverage",
    "test": "jest --coverage",
    "type-check": "tsc"
  },
  "dependencies": {
    "@disruptive-labs/ui": "file:../../vendor/@disruptive-labs/ui",
    "axios": "^0.27.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@testing-library/dom": "^9.3.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/jest": "^29.5.2",
    "@types/react": "^18.2.14",
    "babel-jest": "^29.5.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0",
    "jest-junit": "^16.0.0",
    "react-test-renderer": "^18.2.0",
    "rimraf": "^3.0.2",
    "sass": "^1.63.6",
    "typescript": "^5.2.2"
  },
  "jest": {
    "clearMocks": true,
    "collectCoverageFrom": [
      "<rootDir>/src/**/*.{ts,tsx}"
    ],
    "globalSetup": "<rootDir>/test/setup.js",
    "moduleNameMapper": {
      "^.+\\.module\\.(css|sass|scss)$": "identity-obj-proxy",
      "^.+\\.(css|sass|scss)$": "<rootDir>/test/styleMock.js",
      "^.+\\.(jpg|jpeg|png|gif|webp|svg)$": "<rootDir>/test/fileMock.js",
      "^@test$": "<rootDir>/test/index.tsx"
    },
    "setupFilesAfterEnv": [
      "<rootDir>/test/jest.test.ts"
    ],
    "testEnvironment": "jsdom",
    "testPathIgnorePatterns": [
      "<rootDir>/node_modules/",
      "<rootDir>/test/"
    ],
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": [
        "babel-jest",
        {
          "presets": [
            "next/babel"
          ]
        }
      ]
    },
    "transformIgnorePatterns": [
      "node_modules/(?!(@?disruptive-labs.*|@popperjs/core|es-cookie)/)"
    ]
  },
  "jest-junit": {
    "outputDirectory": "./artifacts/jest",
    "outputName": "results.xml"
  }
}
