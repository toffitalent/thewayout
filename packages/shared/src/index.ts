// NOTE: React components should not exported here

export * from './constants';
export * from './types';
export * from './utils';
