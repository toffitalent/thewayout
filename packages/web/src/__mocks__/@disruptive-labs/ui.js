const actual = jest.requireActual('@disruptive-labs/ui');

const showToast = jest.fn();

actual.Toast.show = showToast;
actual.Toast.danger = showToast;
actual.Toast.info = showToast;
actual.Toast.success = showToast;
actual.Toast.warning = showToast;

module.exports = {
  ...actual,
  showToast,
};
