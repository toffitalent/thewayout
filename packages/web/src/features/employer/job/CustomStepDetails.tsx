import {
  Button,
  Field,
  FieldComponentProps,
  FieldType,
  useForm,
  useWizardContext,
  WizardStep,
} from '@disruptive-labs/ui';
import { WizardData } from './createJobDataTypes';
import { validateStartAt } from './customFieldsValidation';

export interface CustomStepProps extends Omit<Partial<WizardStep>, 'fields'> {
  fields?: FieldComponentProps<FieldType>[];
}

export const CustomStepDetails = (props: CustomStepProps) => {
  const { next, data } = useWizardContext<WizardData>();

  const {
    submitForm,
    control,
    formState: { isSubmitting, isDirty, isValid },
    getValues,
    trigger,
    setError,
    clearErrors,
  } = useForm({
    mode: 'onChange',
    defaultValues: {
      title: data.title || '',
      description: data.description || '',
      department: data.department || '',
      startAtMonth: data.startAtMonth || '',
      startAtYear: data.startAtYear || '',
      typeOfWork: data.typeOfWork || '',
      workingTime: data.workingTime || '',
      numberOfOpenPositions: data.numberOfOpenPositions || '',
    },
    onSubmit: async (values) => {
      await next({
        ...data,
        ...values,
        numberOfOpenPositions: Number(values.numberOfOpenPositions),
      });
    },
  });

  const { fields, nextButtonLabel } = props;
  if (!fields) {
    return null;
  }

  return (
    <form onSubmit={submitForm} className="detailsStep">
      {fields.map((field) => (
        <Field
          {...field}
          key={field.name}
          validate={{
            ...field.validate,
            ...((field.name === 'startAtMonth' || field.name === 'startAtYear') && {
              validate: () => {
                const error = validateStartAt({
                  startAtMonth: getValues('startAtMonth'),
                  startAtYear: getValues('startAtYear'),
                  fieldName: field.name,
                });
                if (error) {
                  setError(field.name as 'startAtMonth' | 'startAtYear', {
                    message: error,
                    type: 'validate',
                  });
                } else {
                  clearErrors(field.name as 'startAtMonth' | 'startAtYear');
                }
                return error;
              },
            }),
          }}
          onChangeCapture={() => {
            if (field.name === 'startAtMonth' || field.name === 'startAtYear') {
              const anotherFieldName =
                field.name === 'startAtMonth' ? 'startAtYear' : 'startAtMonth';
              const secondField = getValues(anotherFieldName);
              if (secondField) {
                trigger(anotherFieldName);
              }
            }
          }}
          control={control}
        />
      ))}
      <Button
        colorScheme="primary"
        fluid
        type="submit"
        maxWidth={96}
        mt={8}
        disabled={isSubmitting || !isValid || (!data.salary && !isDirty)}
      >
        {nextButtonLabel || ''}
      </Button>
    </form>
  );
};
