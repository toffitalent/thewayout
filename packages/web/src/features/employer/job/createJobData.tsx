import { Box, Text, WizardStep } from '@disruptive-labs/ui';
import {
  Experience,
  JobSalaries,
  offenseCategories,
  offenseText,
  states,
  TypeOfWork,
  VeteranOrJustice,
  WorkingTime,
} from '@two/shared';
import { CustomStepAllowedOffense } from '@app/components/CustomStepAllowedOffense';
import { InfoBox } from '@app/components/InfoBox';
import { job, JobEnums } from '@app/data/jobText';
import { months } from '@app/data/months';
import { WizardData } from './createJobDataTypes';
import { CustomStepDetails } from './CustomStepDetails';
import { CustomStepQuestions } from './CustomStepQuestions';
import { CustomStepResponsibilities } from './CustomStepResponsibility';
import { CustomStepSalary } from './CustomStepSalary';

export enum Steps {
  details = 'details',
  location = 'location',
  veteranOrJustice = 'veteranOrJustice',
  offenses = 'offenses',
  responsibilities = 'responsibilities',
  skills = 'skills',
  pay = 'pay',
  questions = 'questions',
}

export const stepsConfig: Record<Steps, { id: string; title: string }> = {
  [Steps.details]: { id: Steps.details, title: 'Add job details' },
  [Steps.location]: { id: Steps.location, title: 'Add job location' },
  [Steps.veteranOrJustice]: {
    id: Steps.veteranOrJustice,
    title: 'Who Can Apply For This Position?',
  },
  [Steps.offenses]: { id: Steps.offenses, title: 'Allowed Offenses' },
  [Steps.responsibilities]: {
    id: Steps.responsibilities,
    title: 'Responsibilities & Expectations',
  },
  [Steps.skills]: { id: Steps.skills, title: 'Skills & Experience' },
  [Steps.pay]: { id: Steps.pay, title: 'Pay & Benefits' },
  [Steps.questions]: { id: Steps.questions, title: 'Job Specific Questions' },
};

const years = Array.from({ length: 3 }, (_, i) => String(new Date().getFullYear() + i));

const getOptionsFromEnum = (enumObj: object) =>
  Object.entries(enumObj).map(([key, value]) => ({
    label: job[key as JobEnums],
    value,
  }));

export const steps: WizardStep[] = [
  {
    ...stepsConfig[Steps.details],
    description:
      'Please refrain from including company name or any employer-identifying information.',
    fields: [
      {
        label: 'Job Title',
        name: 'title',
        type: 'text',
        placeholder: 'E.g. Car Wash Attendant',
        validate: { required: true },
        fluid: true,
      },
      {
        label: 'Short Description',
        name: 'description',
        type: 'text',
        placeholder:
          'E.g. Wash attendants are responsible for performing the daily operational practices of running the car wash facility.',
        validate: { required: true, maxLength: 1000 },
        multiline: true,
        fluid: true,
        helperText: 'Max 1,000 characters.',
      },
      {
        label: 'Department',
        name: 'department',
        type: 'text',
        placeholder: 'E.g. Customer Service',
        validate: { required: true },
        fluid: true,
      },
      {
        name: 'startAtMonth',
        label: 'Start Date',
        type: 'select',
        options: Object.entries(months).map(([value, label]) => ({ label, value })),
        validate: { required: true },
        width: '1/2',
        display: 'inline-block',
        pr: 2,
        className: 'fieldHalfWidth',
        placeholder: 'Select month',
      },
      {
        name: 'startAtYear',
        type: 'select',
        options: years.map((el) => ({ label: el, value: el })),
        validate: { required: true },
        width: '1/2',
        display: 'inline-block',
        pl: 2,
        className: 'fieldHalfWidth',
        placeholder: 'Select year',
      },
      {
        label: 'Work Location',
        name: 'typeOfWork',
        type: 'select',
        options: getOptionsFromEnum(TypeOfWork),
        validate: { required: true },
        placeholder: 'Select',
      },
      {
        label: 'Employment Type',
        name: 'workingTime',
        type: 'select',
        options: getOptionsFromEnum(WorkingTime),
        validate: { required: true },
        placeholder: 'Select',
      },
      {
        label: 'Number of Open Positions',
        name: 'numberOfOpenPositions',
        placeholder: 'E.g. 3',
        type: 'number',
        validate: { required: true, min: 1 },
        fluid: true,
      },
    ],
    className: 'detailsStep',
    nextButtonLabel: 'Save & Next',
    component: <CustomStepDetails />,
    nextStep: (wizardData) =>
      (wizardData as WizardData).typeOfWork === TypeOfWork.onsite
        ? Steps.location
        : Steps.veteranOrJustice,
  },
  {
    ...stepsConfig[Steps.location],
    description: 'Where is the main office / place of work located?',
    fields: [
      {
        label: 'Job Address',
        name: 'address',
        type: 'text',
        placeholder: 'E.g. 7562 Center Ave',
        validate: { required: true },
      },
      {
        label: 'Town / City',
        name: 'city',
        type: 'text',
        placeholder: 'E.g. Los Angeles',
        validate: { required: true },
      },
      {
        label: 'State',
        name: 'state',
        type: 'select',
        options: Object.entries(states).map(([key, value]) => ({ value: key, label: value })),
        placeholder: 'E.g. California',
        validate: { required: true },
      },
      {
        label: 'Postcode / Zipcode',
        name: 'postalCode',
        type: 'text',
        placeholder: 'E.g. 92806',
        validate: { required: true },
      },
    ],
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: Steps.veteranOrJustice,
  },
  {
    ...stepsConfig[Steps.veteranOrJustice],
    description: 'Select all that apply.',
    fields: [
      {
        name: 'veteranOrJustice',
        type: 'choices',
        options: getOptionsFromEnum(VeteranOrJustice),
        multiple: true,
        inline: true,
        validate: (value) => (value.length >= 1 ? undefined : 'Error'),
      },
    ],
    showBackButton: true,
    nextButtonLabel: 'Save & Next',
    nextStep: (wizardData) =>
      (wizardData as WizardData).veteranOrJustice.includes(VeteranOrJustice.justiceImpacted)
        ? Steps.offenses
        : Steps.responsibilities,
  },
  {
    ...stepsConfig[Steps.offenses],
    description:
      'Only candidates that meet your allowable offenses (per State and Federal statutes) can apply for the job listing & will have their profile shared with you.',
    fields: Object.entries(offenseCategories).map(([key, values]) => ({
      name: `offense${key[0].toUpperCase() + key.slice(1)}`,
      label: values.name,
      type: 'choices',
      options: values.categories.map((el) => ({ label: offenseText[el], value: el })),
      inline: true,
      multiple: true,
      className: 'offenseCategoryStep',
    })),
    component: <CustomStepAllowedOffense />,
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: Steps.responsibilities,
  },
  {
    ...stepsConfig[Steps.responsibilities],
    description: 'Add your job responsibilities one at a time. Be mindful of inclusive language.',
    component: <CustomStepResponsibilities />,
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: Steps.skills,
  },
  {
    ...stepsConfig[Steps.skills],
    fields: [
      {
        label: 'Experience',
        name: 'experience',
        type: 'select',
        options: getOptionsFromEnum(Experience),
        validate: { required: true },
        placeholder: 'Select',
      },
      {
        label: (
          <Box display="inline-flex" alignItems="center">
            <Text fontWeight="700" mr={2} mb={0}>
              Skills - Description
            </Text>
            <InfoBox content="Explain the level of experience someone needs to be successful in this job." />
          </Box>
        ),
        name: 'skillsDescription',
        type: 'text',
        multiline: true,
        validate: { required: true, maxLength: 1000 },
        placeholder:
          "E.g. The most successful and happy people have PMA â€“ a Positive Mental Attitude! We are a huge believer in putting attitude above aptitude. We can always teach you how to do the job, but we can't teach a Positive Mental Attitude... You either have it or you don't!",
        helperText: 'Max 1,000 characters.',
      },
    ],
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: Steps.pay,
  },
  {
    ...stepsConfig[Steps.pay],
    fields: [
      {
        label: 'Salary',
        name: 'salary',
        type: 'select',
        options: getOptionsFromEnum(JobSalaries),
        validate: { required: true },
        placeholder: 'Select',
      },
      {
        label: 'Min Salary',
        name: 'min',
        type: 'text',
        validate: { required: true },
        width: '1/2',
        display: 'inline-block',
        pr: 2,
        className: 'fieldHalfWidth',
      },
      {
        label: 'Max Salary',
        name: 'max',
        type: 'text',
        validate: { required: true },
        width: '1/2',
        display: 'inline-block',
        pl: 2,
        className: 'fieldHalfWidth',
      },
      {
        label: (
          <Box display="inline-flex" alignItems="center">
            <Text fontWeight="700" mr={2} mb={0}>
              Pay Description
            </Text>
            <InfoBox content="Can include frequency of pay, when applicants can expect their first pay check, and any other helpful pay information" />
          </Box>
        ),
        name: 'salaryDescription',
        type: 'text',
        multiline: true,
        validate: { required: true, maxLength: 1000 },
        placeholder:
          'E.g. Starting wage is $13 per hour + tips, with an Employer Bonus after positive a 90 Day Review (provided by the employer)',
        fluid: true,
        helperText: 'Max 1,000 characters.',
      },
    ],
    component: <CustomStepSalary />,
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: Steps.questions,
  },
  {
    ...stepsConfig[Steps.questions],
    description: 'Ask qualifying questions that will help identify the most fit candidates.',
    component: <CustomStepQuestions />,
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: 'confirm',
  },
];
