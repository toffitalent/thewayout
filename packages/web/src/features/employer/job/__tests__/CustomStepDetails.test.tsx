import { useWizardContext } from '@disruptive-labs/ui';
import { TypeOfWork, WorkingTime } from '@two/shared';
import { act, fireEvent, render, waitFor } from '@test';
import { Steps, steps } from '../createJobData';
import { CustomStepDetails, CustomStepProps } from '../CustomStepDetails';

const detailStep = steps.find((el) => el.id === Steps.details);

jest.mock('@disruptive-labs/ui', () => ({
  ...jest.requireActual('@disruptive-labs/ui'),
  useWizardContext: jest.fn(),
}));

describe('CustomStepSalary', () => {
  beforeEach(() => {
    (useWizardContext as any).mockImplementation(() => ({
      data: {
        title: 'Senior Marketing Executive',
        description:
          "Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.s",
        department: 'Customer Service',
        typeOfWork: TypeOfWork.hybrid,
        workingTime: WorkingTime.fullTime,
        numberOfOpenPositions: 3,
        startAtYear: '2023',
        startAtMonth: '1',
      },
      setData: jest.fn(),
      goTo: jest.fn(),
    }));
  });

  test('renders correctly', async () => {
    const testDate = new Date('2023-01-26T00:00:00.000Z');
    jest.spyOn(global, 'Date').mockImplementation(() => testDate);

    const { container, findByText } = render(
      <CustomStepDetails fields={(detailStep as CustomStepProps)?.fields || []} />,
    );
    await waitFor(async () => expect(await findByText('Job Title')).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  });

  test('validate start at fields', async () => {
    const testDate = new Date('2023-03-26T00:00:00.000Z');
    jest.spyOn(global, 'Date').mockImplementation(() => testDate);

    const { findAllByRole, getAllByRole } = render(
      <CustomStepDetails fields={(detailStep as CustomStepProps)?.fields || []} />,
    );

    expect((await findAllByRole('button'))[0]).toBeDisabled();

    await act(() =>
      fireEvent.change(getAllByRole('combobox')[1], {
        target: { value: '2025' },
      }),
    );
    await act(() => fireEvent.change(getAllByRole('combobox')[0], { target: { value: '2' } }));
    await waitFor(async () => expect((await findAllByRole('button'))[0]).toBeEnabled());
  });
});
