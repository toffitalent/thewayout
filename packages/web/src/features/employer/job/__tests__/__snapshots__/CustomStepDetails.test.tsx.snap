// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomStepSalary renders correctly 1`] = `
<div>
  <form
    class="detailsStep"
  >
    <div
      class="control"
      role="group"
    >
      <label
        class="label"
        for=":r0:"
      >
        Job Title
      </label>
      <div
        class="root primary md fluid md"
      >
        <input
          class="input"
          id=":r0:"
          name="title"
          placeholder="E.g. Car Wash Attendant"
          type="text"
          value="Senior Marketing Executive"
        />
      </div>
    </div>
    <div
      class="control"
      role="group"
    >
      <label
        class="label"
        for=":r1:"
      >
        Short Description
      </label>
      <div
        class="root primary md fluid md multiline"
      >
        <textarea
          aria-describedby=":r1:-helptext"
          class="input"
          id=":r1:"
          name="description"
          placeholder="E.g. Wash attendants are responsible for performing the daily operational practices of running the car wash facility."
          rows="4"
          type="text"
        >
          Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.s
        </textarea>
      </div>
      <div
        aria-live="polite"
        class="helperText"
        id=":r1:-helptext"
      >
        Max 1,000 characters.
      </div>
    </div>
    <div
      class="control"
      role="group"
    >
      <label
        class="label"
        for=":r2:"
      >
        Department
      </label>
      <div
        class="root primary md fluid md"
      >
        <input
          class="input"
          id=":r2:"
          name="department"
          placeholder="E.g. Customer Service"
          type="text"
          value="Customer Service"
        />
      </div>
    </div>
    <div
      class="control display-inline-block width-1/2 paddingRight-2"
      role="group"
    >
      <label
        class="label"
        for=":r3:"
      >
        Start Date
      </label>
      <div
        class="root primary md fluid md fieldHalfWidth"
      >
        <select
          class="input"
          id=":r3:"
          name="startAtMonth"
        >
          <option
            disabled=""
            hidden=""
            value=""
          >
            Select month
          </option>
          <option
            value="1"
          >
            Jan.
          </option>
          <option
            value="2"
          >
            Feb.
          </option>
          <option
            value="3"
          >
            Mar.
          </option>
          <option
            value="4"
          >
            Apr.
          </option>
          <option
            value="5"
          >
            May
          </option>
          <option
            value="6"
          >
            June
          </option>
          <option
            value="7"
          >
            July
          </option>
          <option
            value="8"
          >
            Aug.
          </option>
          <option
            value="9"
          >
            Sept.
          </option>
          <option
            value="10"
          >
            Oct.
          </option>
          <option
            value="11"
          >
            Nov.
          </option>
          <option
            value="12"
          >
            Dec.
          </option>
        </select>
        <div
          class="icon"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,16c-0.3,0-0.5-0.1-0.7-0.3l-6-6c-0.4-0.4-0.4-1,0-1.4s1-0.4,1.4,0l5.3,5.3l5.3-5.3c0.4-0.4,1-0.4,1.4,0s0.4,1,0,1.4l-6,6C12.5,15.9,12.3,16,12,16z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="control display-inline-block width-1/2 paddingLeft-2"
      role="group"
    >
      <div
        class="root primary md fluid md fieldHalfWidth"
      >
        <select
          class="input"
          id=":r4:"
          name="startAtYear"
        >
          <option
            disabled=""
            hidden=""
            value=""
          >
            Select year
          </option>
          <option
            value="2024"
          >
            2024
          </option>
          <option
            value="2025"
          >
            2025
          </option>
          <option
            value="2026"
          >
            2026
          </option>
        </select>
        <div
          class="icon"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,16c-0.3,0-0.5-0.1-0.7-0.3l-6-6c-0.4-0.4-0.4-1,0-1.4s1-0.4,1.4,0l5.3,5.3l5.3-5.3c0.4-0.4,1-0.4,1.4,0s0.4,1,0,1.4l-6,6C12.5,15.9,12.3,16,12,16z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="control"
      role="group"
    >
      <label
        class="label"
        for=":r5:"
      >
        Work Location
      </label>
      <div
        class="root primary md fluid md"
      >
        <select
          class="input"
          id=":r5:"
          name="typeOfWork"
        >
          <option
            disabled=""
            hidden=""
            value=""
          >
            Select
          </option>
          <option
            value="onsite"
          >
            Onsite
          </option>
          <option
            value="remote"
          >
            Remote
          </option>
          <option
            value="hybrid"
          >
            Hybrid
          </option>
          <option
            value="inPerson"
          >
            In-person
          </option>
        </select>
        <div
          class="icon"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,16c-0.3,0-0.5-0.1-0.7-0.3l-6-6c-0.4-0.4-0.4-1,0-1.4s1-0.4,1.4,0l5.3,5.3l5.3-5.3c0.4-0.4,1-0.4,1.4,0s0.4,1,0,1.4l-6,6C12.5,15.9,12.3,16,12,16z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="control"
      role="group"
    >
      <label
        class="label"
        for=":r6:"
      >
        Employment Type
      </label>
      <div
        class="root primary md fluid md"
      >
        <select
          class="input"
          id=":r6:"
          name="workingTime"
        >
          <option
            disabled=""
            hidden=""
            value=""
          >
            Select
          </option>
          <option
            value="fullTime"
          >
            Full-time
          </option>
          <option
            value="partTime"
          >
            Part-time
          </option>
          <option
            value="contractor"
          >
            Contractor
          </option>
          <option
            value="seasonal"
          >
            Seasonal
          </option>
        </select>
        <div
          class="icon"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,16c-0.3,0-0.5-0.1-0.7-0.3l-6-6c-0.4-0.4-0.4-1,0-1.4s1-0.4,1.4,0l5.3,5.3l5.3-5.3c0.4-0.4,1-0.4,1.4,0s0.4,1,0,1.4l-6,6C12.5,15.9,12.3,16,12,16z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="control"
      role="group"
    >
      <label
        class="label"
        for=":r7:"
      >
        Number of Open Positions
      </label>
      <div
        class="root primary md fluid md"
      >
        <input
          class="input"
          id=":r7:"
          name="numberOfOpenPositions"
          placeholder="E.g. 3"
          type="number"
          value="3"
        />
      </div>
    </div>
    <button
      class="button md solid c-primary disabled fluid maxWidth-96 marginTop-8"
      disabled=""
      type="submit"
    />
  </form>
</div>
`;
